core/base/common/OpenMP.h

```diff
@@ -18,7 +18,7 @@ namespace ttk {
 
   public:
     ParallelGuard(const int nThreads)
-      : oldThreadNumber_{omp_get_num_threads()} {
+      : oldThreadNumber_{omp_get_max_threads()} {
```

Comment:
I believe this will restore the number of threads define by `OMP_NUM_THREADS` regardless of what the variable `threadNumber_` might have been set by the user. to discuss.